src/rewards/patreonAgent.ts

```diff
@@ -0,0 +1,202 @@
+import axios from 'axios';
+import patreonId from '../models/patreonId';
+import { PatreonController } from './patreonController';
+
+interface PatreonDetails {
+  isActivePatreon: boolean;
+  amountCents: number;
+}
+
+class PatreonAgent {
+  private patreonController = new PatreonController();
+
+  public getPatreonAuthorizationUrl() {
+    return this.patreonController.loginUrl;
+  }
+
+  public async getExistingPatreonDetails(
+    usernameLower: string,
+  ): Promise<PatreonDetails> {
+    // This function is to check for features in general on load
+
+    const existingPatreon = await this.getExistingPatreon(usernameLower);
+
+    if (!existingPatreon) {
+      return null;
+    }
+
+    const isActivePatreon = this.hasNotExpired(
+      existingPatreon.pledgeExpiryDate,
+    );
+
+    return { isActivePatreon, amountCents: existingPatreon.amountCents };
+  }
+
+  // This path is hit whenever user clicks link to Patreon button
+  public async registerPatreon(
+    usernameLower: string,
+    code: string,
+  ): Promise<PatreonDetails> {
+    // Check if existing Patreon
+    const existingPatreonDetails = await this.getExistingPatreonDetails(
```

Comment:
Reminder for you to remove this so ppl who upgrade their pledge can re-register.